### PR TITLE
[wptserve] Handle non-ASCII query params correctly in Python 3

### PR DESCRIPTION
**Pull request by Hexcles**
In Python 3, parse_qsl automatically does string decoding and the
default encoding is UTF-8 instead of the isomorphic latin-1 that we
assumed. We need to explicitly set the encoding, similar to
cgi.FieldStorage. A test case is added (which would crash due to
UnicodeError in Python 3 without this fix).

Also add a similar test for POST.

### CODE REVIEW

File: tools/wptserve/tests/functional/test_request.py [hunk 1]
```diff
@@ -117,7 +118,7 @@ def handler(request, response):
     def test_non_ascii_in_headers(self):
         @wptserve.handlers.handler
         def handler(request, response):
-            return request.headers["foo"]
+            return request.headers[b"foo"]
```
Comment (Hexcles) [inline tools/wptserve/tests/functional/test_request.py:121]: This is really a drive-by change to demonstrate the recommended way of accessing headers, even though the getter accepts both binary and text strings.

**Review by stephenmcgruer — APPROVED** [review]
